buildenv/jenkins/docker-slaves/x86/centOS/Dockerfile

```diff
@@ -0,0 +1,219 @@
+    java-1.7.0-openjdk-devel \
+    java-1.8.0-openjdk-devel \
+    lbzip2 \
+    libdwarf-devel \
+    libdwarf \
+    libX11-devel \
+    libXext-devel \
+    libXrender-devel \
+    libXt-devel \
+    libXtst-devel \
+    make \
+    mesa-libGL-devel \
+    mpfr-devel \
+    numactl-devel \
+    ntp \
+    openssl-devel \
+    openssh-server \
+    openssh-clients \
+    perl-Text-CSV \
+    perl-CPAN \
+    perl-DBI \
+    perl-devel \
+    perl-GD \
+    perl-libwww-perl \
+    perl-Time-HiRes \
+    unzip \
+    vim \
+    wget\
+    xz \
+    zip \
+  && rm -rf /var/lib/apt/lists/*
+
+RUN yum -y install https://$(rpm -E '%{?centos:centos}%{!?centos:rhel}%{rhel}').iuscommunity.org/ius-release.rpm \
+  && yum -y install libdwarf-devel
+
+RUN /usr/bin/perl -MCPAN -e'install Text::CSV'
+
+RUN cd /etc/yum.repos.d \
+  && wget http://people.centos.org/tru/devtools-2/devtools-2.repo -O /etc/yum.repos.d/devtools-2.repo \
+  && yum -y install devtoolset-2-gcc devtoolset-2-binutils \
+  && yum -y install devtoolset-2-gcc-c++ devtoolset-2-gcc-gfortran \
+  && scl enable devtoolset-2 bash
+
+RUN mv /opt/rh/devtoolset-2/root/usr/bin/gcc /opt/rh/devtoolset-2/root/usr/bin/gcc-4.8 \
+  && mv /opt/rh/devtoolset-2/root/usr/bin/g++ /opt/rh/devtoolset-2/root/usr/bin/g++-4.8 \
+  && ln -s /opt/rh/devtoolset-2/root/usr/bin/gcc-4.8 /usr/bin/gcc \
+  && ln -s /opt/rh/devtoolset-2/root/usr/bin/g++-4.8 /usr/bin/g++ \
+  && ln -s /opt/rh/devtoolset-2/root/usr/bin/gcc-4.8 /usr/bin/cc
+
+RUN yum -y install curl-devel expat-devel gettext-devel openssl-devel zlib-devel perl-ExtUtils-MakeMaker\
+  && cd /usr/src \
+  && wget https://www.kernel.org/pub/software/scm/git/git-2.5.3.tar.gz \
+  && tar xzf git-2.5.3.tar.gz \
+  && cd git-2.5.3 \
+  && make prefix=/usr/local/git all \
+  && make prefix=/usr/local/git install \
+  && echo 'pathmunge /usr/local/git/bin/' > /etc/profile.d/git.sh \
+  && chmod +x /etc/profile.d/git.sh \
+  && ln -s /usr/local/git/bin/git /usr/bin/git
+
+RUN wget http://www.us.apache.org/dist/ant/binaries/apache-ant-1.10.5-bin.tar.gz \
+  && tar xvfvz apache-ant-1.10.5-bin.tar.gz -C /opt \
+  && rm apache-ant-1.10.5-bin.tar.gz \
+  && ln -s /opt/apache-ant-1.10.5 /opt/ant \
+  && ln -s /opt/ant/bin/ant /usr/bin/ant \
+  && cd /opt/ant/lib \
+  && wget https://sourceforge.net/projects/ant-contrib/files/ant-contrib/1.0b3/ant-contrib-1.0b3-bin.tar.gz \
+  && tar xvfvz ant-contrib-1.0b3-bin.tar.gz -C .\
+  && cd ant-contrib \
+  && mv ant-contrib-1.0b3.jar .. \
+  && rm -rf ../ant-contrib
+
+RUN cd /usr/src/ \
+  && wget http://developer.download.nvidia.com/compute/cuda/7.5/Prod/local_installers/cuda_7.5.18_linux.run \
+  && chmod +x cuda_7.5.18_linux.run \
+  && /usr/src/cuda_7.5.18_linux.run --silent --toolkit --override
+
+# add user home/jenkins
+RUN useradd -ms /bin/bash "${HOMEDIR}" \
+  && mkdir /home/"${HOMEDIR}"/.ssh/ \
+  && COPY authorized_keys /home/jenkins/.ssh/authorized_keys \
+  && chown -R "${HOMEDIR}":"${HOMEDIR}" /home/"${HOMEDIR}"
```

Comment:
I think the braces around the ENV variables are optional
https://docs.docker.com/engine/reference/builder/#environment-replacement
```
ENV foo /bar
WORKDIR ${foo}   # WORKDIR /bar
ADD . $foo       # ADD . /bar
COPY \$foo /quux # COPY $foo /quux
```
But I don't think the quotes are required